test/timesheet.js

```diff
@@ -0,0 +1,223 @@
+const db = require('../source/database');
+const Timesheet = require('../source/timesheet');
+const collectionName = 'Timesheets';
+
+describe('Timesheet.js Tests:', function(){
+    
+    // Connects to database before testing
+    before(async function() {
+        await db.myConnect();
+    });
+
+    // Closes database connection after testing
+    after(async function(){
+        await db.myClose();
+    });
+
+    describe('Timesheet Validation Tests:', function(){
+
+        let timesheetValues = {
+            _id : db.createMongoID(),
+            date : new Date(),
+            employeeID : db.createMongoID(),
+            jobNumber : 12345,
+            jobID : db.createMongoID(),
+            estCrewSize : 3,
+            estCrewHours : 8,
+            tmCrewSize : 0,
+            tmCrewHours : 0,
+            lunchTaken : true,
+            jobFinished : true,
+            offHauled : false,
+            yardsHauled : 0,
+            notes : 'Test Notes'
+        };
+
+        it('Timesheet Validation - Happy Path', async function(){
+
+            let timesheet = new Timesheet(timesheetValues);
+            await timesheet.validate();
+
+        });
+
+        it('Timesheet Validation - _id', async function(){
```

Comment:
All of these tests will pass with a false positive if no error is thrown at all.

Reply:
Fixed by throwing a new error ('Error expected but not thrown in Timesheet Validation - [Test]') after the timesheet.validate() call that will only be executed if no error is thrown.